src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -263,17 +263,13 @@ internal static string GetRuntimeDirectoryOpt()
         /// The IsConnected property on named pipes does not detect when the client has disconnected
         /// if we don't attempt any new I/O after the client disconnects. We start an async I/O here
         /// which serves to check the pipe for disconnection.
-        ///
-        /// This will return true if the pipe was disconnected.
         /// </summary>
-        private static async Task CreateMonitorDisconnectTask(
-            NamedPipeClientStream pipeStream,
-            CancellationToken cancellationToken)
+        internal static async Task CreateMonitorDisconnectTask(
+            PipeStream pipeStream,
+            string identifier = null,
```

Comment:
Nit: why make this parameter optional? If only two callers, then we can pass "client" and "server" explicitly.